src/preact.d.ts

```diff
@@ -11,7 +11,8 @@ declare namespace preact {
 	interface PreactHTMLAttributes {
 		dangerouslySetInnerHTML?:DangerouslySetInnerHTML;
 		key?:string;
-		ref?:(el?: Element) => void;
+		defaultValue?:string;
+		ref?:<T,S>(el:Component<T,S>|Element)=>void;
```

Comment:
Ah, no you're right - sorry I missed that! Indeed, an Element VNode will only ever invoke its ref with either an `Element` or `null`.